src/signature.c

```diff
@@ -157,6 +158,63 @@ GBytes *cms_sign(GBytes *content, const gchar *certfile, const gchar *keyfile, G
 	return res;
 }
 
+gchar* get_pubkey_hash(X509 *cert) {
+		gchar *data = NULL;
+		BIO *bio;
+		unsigned char *ibuf = NULL;
+		unsigned char *buf2;
+		unsigned int len = 0;
+		unsigned int n = 0;
+		unsigned char md[SHA256_DIGEST_LENGTH];
+
+		bio = BIO_new(BIO_s_mem());
+
+		len = i2d_X509_PUBKEY(X509_get_X509_PUBKEY(cert), NULL);
+		buf2 = ibuf = g_malloc(len);
+		if (ibuf) i2d_X509_PUBKEY(X509_get_X509_PUBKEY(cert), &buf2);
+
+		g_assert(buf2 - ibuf == len);
+
+		if (!EVP_Digest(ibuf, len, md, &n, EVP_sha256(), NULL)) {
+			g_warning("Error in EVP_Digest\n");
+			goto out;
+		}
+
+		for (int j = 0; j < (int)n; j++) {
```

Comment:
Maybe use g_string_sized_new(), g_string_append_printf() and g_string_truncate instead of a BIO and the ?: ?

Reply:
`g_string_sized_new()` sounds good. Will save some LOC.